src/common/components/manage-auth-icon/index.tsx

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import { History } from "history";
+import { PrivateKey, PublicKey } from "@hiveio/dhive";
+
+import { actionType } from "../manage-authority/types";
+import DropDown, { MenuItem } from "../dropdown";
+
+import { KebabMenu, revokeSvg, copyOutlinSvg, keyOutlineSvg } from "../../img/svg";
+import "./index.scss";
+
+interface Props {
+  type: string;
+  account?: string;
+  history: History | null;
+  Pkey?: PrivateKey | string | PublicKey;
+  label?: string;
+  action?: string;
+  keyType?: string;
+  onRevoke?: (account: string) => void;
+  onCopy?: (text: string) => void;
+  onImport?: (type: string) => void;
+  onReveal?: () => void;
+}
+
+const ManageAuthIcon = (props: Props) => {
+  const revokeClicked = () => {
+    const { onRevoke, account } = props;
+    if (onRevoke) {
+      onRevoke(account!);
+    }
+  };
+
+  const copyClicked = () => {
+    const { onCopy, Pkey, label } = props;
+    if (onCopy) {
+      onCopy(Pkey!.toString());
+    }
+  };
+
+  const keysHandleClicked = () => {
+    const { onImport, action, keyType, onReveal } = props;
+    if (action === actionType.Import && onImport) {
+      onImport(keyType!);
+    } else {
+      if (onReveal) {
+        onReveal();
+      }
+    }
+  };
+
+  let menuItems: MenuItem[] = [];
+  switch (props.type) {
+    case actionType.Revoke:
+      menuItems = [
+        {
+          label: "Revoke",
```

Comment:
@MTSaeed these labels should be translatable if part of UI.

Reply:
@feruzm Updated.